### PR TITLE
Feat/compute hit color

### PR DESCRIPTION
**Pull request by mathieurobert1**
Compute hit colors based on the primitive's material (flat color currently) and the lights.

I created this three lights computations: Ambient, Punctual and Directional.

More details : #36

### CODE REVIEW

**Review by theophile-jr — DISMISSED** [review]

File: include/Lights/Directional.hpp [hunk 1]
```diff
@@ -43,42 +43,42 @@ class Light::Directional : public Light::ILight {
...
          * @param position New position of Directional Light
          */
         void setPosition(Math::Point3D position);
 
         /**
          * @brief Get the Direction number of Point light
-         * 
-         * @return double 
+         *
+         * @return double
```
Comment (Njord201) [inline include/Lights/Directional.hpp:61]: Please fix the return type in documentation

File: include/Lights/Point.hpp [hunk 2]
```diff
@@ -42,28 +42,28 @@ class Light::Point : public Light::ILight {
 
         /**
          * @brief Get the Position number of Point light
-         * 
-         * @return double 
+         *
+         * @return double
          */
         Math::Point3D getPosition(void) const;
```
Comment (Njord201) [inline include/Lights/Point.hpp:48]: Please fix the return type in documentation

File: src/Plugins/Primitives/Plane/Plane.cpp [hunk 3]
```diff
@@ -58,3 +58,10 @@ Math::Point3D Primitive::Plane::hitPoint(const Raytracer::Ray &ray) const
     }
     return rayDirection * t;
 }
+
+Math::Point3D Primitive::Plane::computeColor(const Math::Point3D& hitPoint, const Light::LightsContainer& lights) const
```
Comment (Njord201) [inline src/Plugins/Primitives/Plane/Plane.cpp:62]: Create an Issue an a commentary for TODO

**Review by Njord201 — CHANGES_REQUESTED** [review]

File: include/Lights/Directional.hpp [hunk 4]
```diff
@@ -43,42 +43,42 @@ class Light::Directional : public Light::ILight {
 
         /**
          * @brief Get the Position number of Point light
-         * 
-         * @return double 
+         *
+         * @return double
          */
         Math::Point3D getPosition(void) const;
```
Comment (Njord201) [inline include/Lights/Directional.hpp:49]: Same herek, the type of return is not the same in documentation

**Review by Njord201 — CHANGES_REQUESTED** [review]

**Review by Njord201 — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

**Pull request merged**